synthetic_data_benchmark/synthesizer/tgan_synthesizer.py

```diff
@@ -304,67 +302,88 @@ def train(self, train_data):
         self.cond_generator = Cond(train_data, self.transformer.output_info)
 
         generator= Generator(self.embeddingDim + self.cond_generator.n_opt, self.genDim, data_dim).to(self.device)
-        discriminator = Discriminator(data_dim, self.disDim).to(self.device)
+        discriminator = Discriminator(data_dim + self.cond_generator.n_opt, self.disDim).to(self.device)
 
         optimizerG = optim.Adam(generator.parameters(), lr=2e-4, betas=(0.5, 0.9), weight_decay=self.l2scale)
         optimizerD = optim.Adam(discriminator.parameters(), lr=2e-4, betas=(0.5, 0.9))#, weight_decay=self.l2scale)
 
         max_epoch = max(self.store_epoch)
         assert self.batch_size % 2 == 0
-        mean = torch.zeros(self.batch_size//2, self.embeddingDim, device=self.device)
+        mean = torch.zeros(self.batch_size, self.embeddingDim, device=self.device)
         std = mean + 1
 
 
         steps_per_epoch = len(train_data) // self.batch_size
         for i in range(max_epoch):
             for id_ in range(steps_per_epoch):
-                real = data_sampler.sample(self.batch_size)
-                real = torch.from_numpy(real.astype('float32')).to(self.device)
-                y_real = discriminator(real)
-
                 fakez = torch.normal(mean=mean, std=std)
-                fakez = torch.cat([fakez, fakez], dim=0)
 
                 condvec = self.cond_generator.generate(self.batch_size)
                 if condvec is None:
-                    pass
+                    c1, m1, col, opt = None, None, None, None
+                    real = data_sampler.sample(self.batch_size, col, opt)
                 else:
-                    c1, m1 = condvec
+                    c1, m1, col, opt = condvec
                     c1 = torch.from_numpy(c1).to(self.device)
                     m1 = torch.from_numpy(m1).to(self.device)
                     fakez = torch.cat([fakez, c1], dim=1)
-                fake = generator(fakez)
 
+                    perm = np.arange(self.batch_size)
+                    np.random.shuffle(perm)
+                    real = data_sampler.sample(self.batch_size, col[perm], opt[perm])
+                    c2 = c1[perm]
+
+                fake = generator(fakez)
                 fakeact = apply_activate(fake, self.transformer.output_info)
-                y_fake = discriminator(fakeact)
+
+
+                real = torch.from_numpy(real.astype('float32')).to(self.device)
+
+                if c1 is not None:
+                    fake_cat = torch.cat([fakeact, c1], dim=1)
+                    real_cat = torch.cat([real, c2], dim=1)
+                else:
+                    real_cat = real
+                    fake_cat = fake
+
+
+                # print(real_cat[0])
+                # print(fake_cat[0])
+                # assert 0
```

Comment:
If the commentged lines are not required, delete them along the excess of blank lines